src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/EmbeddedLanguages/VirtualChars/CSharpVirtualCharService.cs

```diff
@@ -145,6 +161,72 @@ private static VirtualCharSequence TryConvertRawStringToVirtualChars(SyntaxToken
             return CreateVirtualCharSequence(tokenText, offset, startIndexInclusive, endIndexExclusive, result);
         }
 
+        private static VirtualCharSequence TryConvertMultiLineRawStringToVirtualChars(SyntaxToken token)
+        {
+            // The containing expression for this token.  This is needed so that we can determine the indentation
+            // whitespace based on the last line of the containing multiline literal.
+            ExpressionSyntax parentExpression = token.Kind() is SyntaxKind.MultiLineRawStringLiteralToken or SyntaxKind.UTF8MultiLineRawStringLiteralToken
+                ? (LiteralExpressionSyntax)token.GetRequiredParent()
+                : (InterpolatedStringExpressionSyntax)token.GetRequiredParent().GetRequiredParent();
+
+            // Whether or not the token contains the delimiters, or if the parent owns them.  This will affect if we
+            // need to skip looking at the start/end line of the token.
+            var hasDelimiters = parentExpression is LiteralExpressionSyntax;
+
+            // if this is the first text content chunk of the multi-line literal.  The first chunk contains the leading
+            // indentation of the line it's on (which thus must be trimmed), while all subsequent chunks do not (because
+            // they start right after some `{...}` interpolation
+            var isFirstChunk =
+                parentExpression is LiteralExpressionSyntax ||
+                (parentExpression is InterpolatedStringExpressionSyntax { Contents: var contents } && contents.First() == token.GetRequiredParent());
+
+            if (parentExpression.GetDiagnostics().Any(d => d.Severity == DiagnosticSeverity.Error))
+                return default;
+
+            // Use the parent multi-line expression to determine what whitespace to remove from the start of each line.
+            var parentSourceText = parentExpression.SyntaxTree.GetText();
+            var indentationLength = parentSourceText.Lines.GetLineFromPosition(parentExpression.Span.End).GetFirstNonWhitespaceOffset() ?? 0;
+
+            // Create a source-text view over the token.  This makes it very easy to treat the token as a set of lines
+            // that can be processed sensibly.
+            var tokenSourceText = SourceText.From(token.Text);
+
+            // If we're on the very first chunk of the multi-line raw string literal, then we want to start on line 1 so
+            // we skip the space and newline that follow the initial `"""`.
+            var startLineInclusive = hasDelimiters ? 1 : 0;
```

Comment:
Not exactly sure what I want to say here - but it seems weird that this has to account for the idea that the multi-line raw string contents are defined only on lines in between the `"""` (and not the lines containing the `"""`).  Is it not possible for that logic to be pushed up somewhere?  E.g. you have a token representing the string, get its contents